build_resolvers/lib/src/analysis_driver_filesystem.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:analyzer/source/file_source.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/clients/build_resolvers/build_resolvers.dart';
+import 'package:build/build.dart' hide Resource;
+import 'package:path/path.dart' as p;
+
+/// The in-memory filesystem that is the analyzer's view of the build.
+///
+/// Tracks modified paths, which should be passed to
+/// `AnalysisDriver.changeFile` to update the analyzer state.
+class AnalysisDriverFilesystem implements UriResolver, ResourceProvider {
+  final Map<String, String> _data = {};
+  final Set<String> _changedPaths = {};
```

Comment:
```suggestion
  final _data = <String, String>{};
  final _changedPaths = <String>{};
```

Does anyone else prefer this style ☝️

Reply:
Personally not for fields. Declaring types on the left allow them to be consistent with getters and instance methods, and require me to consciously specify types for public members. These aren't public but it would be strange to be inconsistent between public and private fields.